### PR TITLE
W-10819010 - Button to Import Selected Data Import records

### PR DESCRIPTION
**Pull request by callen-sfdo**
WI: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000sNlO5YAK/view

-  Adds a new button to the data import list view which allows the user to select a number of non-imported data import records and only import those records.
- Utilizes a new VF page and controller to validate the selected records then redirect them to the existing BDI_DataImport page for configuration and processing of those records.
- Existing BDI_DataImport_CTRL modified to allow for processing of specific batch, specific set of data import ids, or process all records if neither is provided.
- Added and updated labels in the new controller, page, and in existing BDI_DataImport_API class that had hardcoded labels.
- Renamed existing 'Start Data Import' button to be consistent with new button
- Modified list view buttons to exclude some additional unneeded buttons and reorder the remaining ones.

-  Adds a new 'Import Selected Data Import Records' button which can be placed on the Data Import list view and allows the user to select a number of non-imported data import records and only import those records.

# Critical Changes

# Changes
- In NPSP Data Importer we added the ability to process selected records in a list view. To use this new functionality, add the Import Selected Data Import Records button to your NPSP Data Import list views.

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata
- Actions and Buttons
  - Import Selected Data Import Records

# Deleted Metadata

### CODE REVIEW

**Comment by callen-sfdo** [comment]
@melissabarber Please review the button changes as well as the new custom labels. Please let me know if you like a short meeting to discuss where they are used.

**Comment by callen-sfdo** [comment]
@lparrott Not sure who should review this on the NPSP side now that Prince and Ken are gone, please review or let me know if there is someone else who should review.

**Comment by callen-sfdo** [comment]
@mldyang @lparrott FYI, I know I don't have code coverage yet on the new controller, just wanted to make this available for review while I worked on that new test class.

File: force-app/main/default/classes/BDI_DataImportCTRL_TEST.cls [hunk 1]
```diff
@@ -162,6 +160,172 @@ private with sharing class BDI_DataImportCTRL_TEST {
...
+
+        // This contact will cause the Data Import record to match on it
+        Contact testContA = new Contact(FirstName = 'Susie',
+                                         LastName = 'TestGroupA01',
+                                        npe01__HomeEmail__c = 'testgroupAcontact01Personal@fakedata.com',
+                                        Email = 'testgroupAcontact01Personal@fakedata.com');
+        insert testContA;
+
+        // This DI will the be passed into the controller and processed
+        DataImport__c testDataImportA = new DataImport__c(
```
Comment (mldyang) [inline force-app/main/default/classes/BDI_DataImportCTRL_TEST.cls:181]: Would it make sense to put Data Import record creation into a test util class of some sort?

**Review by mldyang — APPROVED** [review]
looks good!

File: force-app/main/default/classes/BDI_ProcessSelectedDIs_CTRL.cls [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    private Integer maxNumOfDi = 100;
+    public Boolean error {get; set;}
+
+    /*********************************************************************************************************
+    * @description Constructor that extracts the Data Import Ids from the URL and performs some checks on them
+    * before reformatting the ids for the redirect.
+    */
+    public BDI_ProcessSelectedDIs_CTRL(ApexPages.StandardSetController controller) {
+
+        DataImport__c[] diRecs = (List<DataImport__c>)controller.getSelected();
```
Comment (voduyemi) [inline force-app/main/default/classes/BDI_ProcessSelectedDIs_CTRL.cls:49]: Can we break up the responsibilities of this constructor, we seem to have 2 distinct responsibilities here breaking it up which will help us unit test the different behaviours.

**Review by voduyemi — CHANGES_REQUESTED** [review]
Thanks, Charles for this work, I've got some comments around security and breaking up methods. Please let me know if you'd like to chat about it.

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 3]
```diff
@@ -5007,6 +5023,38 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <shortDescription>bdiSelectNoDIRecordsSelected</shortDescription>
+        <value>No Data Import records were selected for processing.</value>
+    </labels>
+    <labels>
+        <fullName>bdiSelectReturnToListView</fullName>
+        <categories>Data Importer</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>bdiSelectReturnToListView</shortDescription>
+        <value>Return to List View</value>
```
Comment (melissabarber) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:5048]:
```suggestion
        <value>Go Back</value>
```

**Review by melissabarber — COMMENTED** [review]
@callen-sfdo refined the labels with the team, and the updates are ready to go. Let me know if you have any questions.

File: force-app/main/default/classes/BDI_DataImportCTRL_TEST.cls (hunk 1 above)
Comment (callen-sfdo) [reply force-app/main/default/classes/BDI_DataImportCTRL_TEST.cls:181]: I think thats a good idea, but I think I would like to hold off at this time.

**Review by mldyang — APPROVED** [review]
looks good!

**Review by voduyemi — APPROVED** [review]
Looks good! Thanks @callen-sfdo

**Pull request merged**